### ISSUE TITLE
Blob trigger stopped working in Function App

### ISSUE THREAD

**Issue by TimMarshAU** [Azure/azure-functions-host#8580]
Hi all

Sorry if in wrong place - please let me know where this should go.

We're using blob triggers in a Function App to detect new files.

Function Extension Version ~4, Node.js 16 LTS. 

Expected behaviour: new file added to container/path triggers the Function App. 

Problem:
Function App + blob triggers has been working fine for around 2 months until 25/7/2022 AEST when triggers (2 blob triggers in 2 functions under the 1 app) just stopped working.

Steps to reproduce:
Drop a file in container and blob trigger never triggers.

Debug: 
Weirdly, I can run the functions from VS Code just fine.

Checking Application Insights, I see an error: Missing workitem Storage Queues: azure-webjobs-blobtrigger-<functionappname>

However, the queue is definitely there.

Upon raising ticket with MSFT Unified Support, it was noted that there was a platform upgrade from version 4.5.1.18321 to 4.7.2.18626 at almost exactly the same time.

We were advised to update our host.json file from the below (bolding mine).

`{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Request"
      }
    }
  },
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle",
    "version": "[**2.*, 3.0.0)"**
  }
}`

to change the extensionBundle version to "version": "[3.3.0, 4.0.0)" - I'll be doing that tomorrow.

I've gone back to MSFT with some questions.

Weirdly, after logging in to Azure Portal tonight I can see our Function App version has changed to ~3 and Node to 12, changes which we didn't make.

**Comment by Ved2806** [comment]
Hi @TimMarshAU Thank you for the feedback. we will investigate this further and let you know about the findings soon!

**Comment by TimMarshAU** [comment]
> Hi @TimMarshAU Thank you for the feedback. we will investigate this further and let you know about the findings soon!

Solved!

We terminated our storage account on a private IP and aliased this to the storage account URI.

We then disallowed public internet access on our NSG (all our workloads are in a VNET).

However, we didn't realise the storage account sub-resources (queues endpoint etc) remained on a public IP, ie they needed a private endpoint.

So the NSG rule blocked access to the queues where the triggers were hitting.

We allowed the NSG rule and service restored.

So we have added private IPs and aliasing on the queues endpoint (but not table, fileshare etc).

T
